src/System.Security.Cryptography.Csp/tests/CspParametersTests.cs

```diff
@@ -33,5 +34,103 @@ public static void SetFlags_ValidatesInput()
             Assert.Throws<ArgumentException>(
                 () => cspParameters.Flags = (CspProviderFlags)0x0100 | CspProviderFlags.NoPrompt);
         }
+
+        [Fact]
+        public static void KeyPassword_SetGet()
+        {
+            var cspParameters = new CspParameters();
+            using (var pwd = new SecureString())
+            {
+                pwd.AppendChar('p');
+                cspParameters.KeyPassword = pwd;
+                Assert.Same(pwd, cspParameters.KeyPassword);
+            }
+        }
+
+        [Theory(Skip = "Manual test - requires Smart Card - read instructions")]
+        [InlineData(true)]
+        [InlineData(false)]
+        public static void KeyPassword_SmartCard_Manual_Test(bool correctPassword)
```

Comment:
@bartonjs I think it is worth to keep manual test in case we get any issue with this in the future - it saves at least a little bit of time investigating how to test this